### DISCUSSION TITLE
Question on AllocationPool::newRunImpl implementation

### DISCUSSION THREAD

**Discussion by mikechen3** [facebookincubator/velox#3355] [Q&A]
https://github.com/facebookincubator/velox/blob/babb1c34f1653068fdb45a1a45c273fabb1fc43d/velox/common/memory/AllocationPool.cpp#L62-L80

Problem statement:
It seems in AllocationPool::newRunImpl, we try to point the `currentRun_` variable to a run whose size is greater or equal than `numPages` parameter. However it seems when we don't enter the `if (currentRun_ >= allocation_.numRuns()) {` condition, we simply move the `currentRun` to point to the next run in `allocation_`.

Question:
How does next run in `allocation_` guarantee to have size GE than `numPages`?

**Comment by mbasmanova** [comment]
@xiaoxmeng Meng, can you help answer this question?
